include/zephyr/net/coap.h

```diff
@@ -1164,6 +1164,32 @@ int coap_resource_notify(struct coap_resource *resource);
  */
 bool coap_request_is_observe(const struct coap_packet *request);
 
+/**
+ * @brief CoAP transmission parameters.
+ */
+struct coap_transmission_parameters {
+	/** Maximum number of retransmissions. */
+	uint8_t max_retransmission;
+	/**  Initial AKC timeout. Value is used as a base value to retry pending CoAP packets. */
+	uint32_t ack_timeout;
+	/** Set CoAP retry backoff factor. A value of 200 means a factor of 2.0. */
+	uint16_t coap_backoff_percent;
```

Comment:
We could adjust the field ordering here, now the size of this is 12 bytes, but with
```
	/**  Initial AKC timeout. Value is used as a base value to retry pending CoAP packets. */
	uint32_t ack_timeout;
	/** Set CoAP retry backoff factor. A value of 200 means a factor of 2.0. */
	uint16_t coap_backoff_percent;
	/** Maximum number of retransmissions. */
	uint8_t max_retransmission;
```
it drops to 8 bytes.

Reply:
Order changed. Thanks